### PR TITLE
Bugfixes in User Module

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Fix: #1971 

Bugs are elaborated in the below document with the module index as the User.
https://docs.google.com/spreadsheets/d/1CB139UvzvUtB1JOHYP0nmEqjxdoHQPGWLGjD_niH9pw/edit#gid=1379931937

### CODE REVIEW

File: tests/Unit/Admin/User/ClientControllerTest.php [hunk 1]
```diff
@@ -95,4 +95,23 @@ public function test_getBaseQueryForUserSearch_GivesPhoneNumberFormattedWithCoun
...
+            'company'=> 'demo',
+            'country'=>'IN',
+            'email'=>'test@test.com',
+            'state' => 'karnataka',
+            'timezone_id' => '79',
+            'mobile' => '9898789887',
+            'mobile_code' => '91',
+
+        ]);
+        $this->assertDatabaseHas('users', ['email'=>'test@test.com']);
```
Comment (ajaz-ul-haq) [inline tests/Unit/Admin/User/ClientControllerTest.php:115]: need to assert for the response too

Comment (sowmiyasubramaniyan) [reply tests/Unit/Admin/User/ClientControllerTest.php:117]: Done

**Review by ajaz-ul-haq — DISMISSED** [review]

**Review by priyadarshini2101 — APPROVED** [review]
Tested the pr in server instance.

- [ ] In All Users, Under Advanced Search, the Country dropdown is not disappearing either after selecting the particular country by clicking or by entering the action. Until click outside the dropdown.

After fix
feb6cou


- [ ]  The page is Refreshing Without selecting the user to Suspend.

After fix
feb6ref

- [ ] Once the user is created, there is no option to change the country.

After fix
feb6scou

- [ ]  Accepting invalid inputs in Zip/Postal Code field, while creating a user.

After fix
feb6zip

**Review by SarfrazS — APPROVED** [review]
Mentioned issues are fixed.

**Pull request merged**